src/enumerable.cr

```diff
@@ -493,6 +493,28 @@ module Enumerable(T)
     nil
   end
 
+  # Returns the last element in the collection. Raises `Enumerable::EmptyError`
+  # if the collection is empty.
+  def last
+    found = false
+    obj = uninitialized T
```

Comment:
I don't understand, are we admitting that it's OK to use `uninitialized` in high-level code?

Reply:
It could be seen as high-level code. It can also be seen as a fundamental thing in the stdlib. I know Ruby does all kind of things in C in their stdlib code, so maybe that's fine?

That said... do you think there's a way to implement this without `uninitialized`?

Reply:
Well, I don't know about that. It is being used there anyway, so if you think that's wrong please open a separate issue about that, thanks.

Reply:
> That said... do you think there's a way to implement this without `uninitialized`?

@asterite I can think of one at least, but given there's plenty of other places within `Enumerable` where this pattern is being used, I don't think we should go into some obscure mambo-jumbo just for an abstract reasoning - whether or not it is, or it is not _OK_ - implementation is safe & simple, what's wrong with that?

Reply:
Could you not just do like:
```cr
def last
  obj = nil
  each do |elem|
    obj = elem
  end
  raise Enumerable::EmptyError.new if obj.nil?
  obj
end
```

Reply:
This doesn't understand containers which can actually have `nil`

Reply:
That's exactly the problem. I guess we can have a marker type which isn't expected to be in any collection. For example:

```crystal
module Enumerable(T)
  record Dummy

  def last
    obj = Dummy
    each { |e| obj = e }
    raise Enumerable::EmptyError.new if obj.is_a?(Dummy.class)
    obj
  end
end
```

What do you think?

Not that we want to do it here, but maybe for a later iteration. Maybe we could even make it a public type and explain how it can't be used. I think we could use it in `Channel` too, and many other places where `uninitialized` is used. Then we can almost remove `uninitialized` from the language, or just keep it for `StaticArray`. Then nobody will get crashes because of misuse of `uninitialized`.

Reply:
@asterite Sounds good, we already have a `Iterator::Stop` which kinda "paves the way" serving similar purpose.

Reply:
Alternative implementation would be a separate flag variable.